client/src/bastionlab/client.py

```diff
@@ -73,26 +94,42 @@ def __call__(self, _, callback):
 
 @dataclass
 class Connection:
+    """
+    This class represents a connection to a remote server. It holds the necessary
+    information to establish and manage the connection, such as the hostname, port,
+    identity (signing key), and token (if applicable).
+
+    Attributes:
+        host (str): The hostname or IP address of the remote server.
+        port (int, optional): The port to use for the connection. Defaults to 50056.
+        identity (SigningKey, optional): The signing key to use for authentication.
+            If not provided, the connection will not be authenticated.
+        channel (Any): The underlying channel object used to send and receive messages.
```

Comment:
Maybe it's worth mentioning that the user does not have to generate a channel by himself